File: Celebratorica/app/src/main/java/io/blushine/celebratorica/settings/SettingsRepo.java
package io.blushine.celebratorica.settings;

import io.blushine.android.util.Time;

/**
 * Repository for settings
 */
class SettingsRepo {
private static final String TAG = SettingsRepo.class.getSimpleName();
private static SettingsRepo mInstance = null;
private SettingsPrefsGateway mPrefsGateway = new SettingsPrefsGateway();

/**
 * Enforces singleton pattern
 */
private SettingsRepo() {
}

/**
 * Get singleton instance
 * @return get instance
 */
public static SettingsRepo getInstance() {
	if (mInstance == null) {
		mInstance = new SettingsRepo();
	}
	return mInstance;
}

/**
 * Get the reminder (notification) time
 * @return reminder (notification) time
 */
Time getReminderTime() {
	return mPrefsGateway.getReminderTime();
}
}

File: Celebratorica/app/src/main/java/io/blushine/rmw/settings/SettingsRepo.kt
package io.blushine.rmw.settings

import io.blushine.android.common.Time

/**
 * Repository for settings
 */
object SettingsRepo {
	/**
	 * Get the reminder (notification) time
	 * @return reminder (notification) time
	 */
	fun getReminderTime(): Time {
		return SettingsPrefsGateway.getReminderTime()
	}

	/**
	 * @return current storage location
	 */
	fun getStorageLocation(): StorageLocations {
		return SettingsPrefsGateway.getStorageLocation()
	}

	/**
	 * Set the storage location
	 * @param storageLocation the location to store the items
	 */
	internal fun setStorageLocation(storageLocation: StorageLocations) {
		SettingsPrefsGateway.setStorageLocation(storageLocation)
	}
}
